modules/github/standardized-repository-settings/main.tf

```diff
@@ -0,0 +1,145 @@
+/**
+  * This module sets up a GitHub repository in a standardized way, to ensure
+  * consistency among things like branch and PR handling, labels and so on.
+  */
+
+terraform {
+  required_version = "0.15.3"
+
+  required_providers {
+    github = {
+      source  = "integrations/github"
+      version = "4.18.0"
+    }
+  }
+}
+
+provider "github" {
+  owner = "XenitAB"
+  token = var.github_token
+}
+
+resource "github_repository" "this" {
+  name        = var.repository_name
+  description = var.repository_description
+
+  allow_auto_merge       = false
+  allow_merge_commit     = true
+  allow_rebase_merge     = true
+  allow_squash_merge     = true
+  delete_branch_on_merge = true
+  has_downloads          = false
+  has_issues             = true
+  has_wiki               = false
+  visibility             = var.repository_visibility
+  vulnerability_alerts   = var.vulnerability_alerts
+}
+
+resource "github_branch_protection" "this" {
```

Comment:
That's covered by this option, intentionally left blank since it's default:
dismiss_stale_reviews: (Optional) Dismiss approved reviews automatically when a new commit is pushed. Defaults to false.